### DISCUSSION TITLE
Creating tables at runtime

### DISCUSSION THREAD

**Discussion by CPoncet** [SeaQL/sea-orm#916] [Q&A] [answered]
Hello,

I need to build a very specific database structure that will change during time. I need dynamic table naming and ability to change table attributes at runtime.

In Sea ORM docs I could only figure one way of doing this : via raw SQL. Is there an alternative?
I am pretty sure it is not a good design so if you have any recommandation, that would be perfect!

For example, I am considering an EAV pattern but I am concerned about performance and it would not allow users to export / import clean data from the database.

Thank you!

**Comment by billy1624** [comment, marked as answer]
Hey @CPoncet, thanks for asking! We actually have a project based on SeaORM that has dynamic schema structure and creating schema during run-time. You might want to check it out: https://github.com/SeaQL/starfish-ql

**Reply by CPoncet** [reply]
Looks very promising!

It could be the database abstraction I need. Too bad it is supporting only MySQL for now, I would like to use PostgreSQL instead.
